### PR TITLE
feat: add support for Redis Stream

### PR DESCRIPTION
**Pull request by raphael0202**
Implement support for Redis stream, that are replacing Redis queues.

This comes with a full redesign of the update mechanism, and unit tests to test the code.
Coverage is not full (61% on _import.py), additional tests will be needed to reach 100%.

### CODE REVIEW

File: app/_import.py [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+        # We update the min_id to the last ID of the batch
+        min_id = f"({batch[-1][0]}"
+        for timestamp_id, item in batch:
+            id_ = item[id_field_name]
+            logger.debug("Fetched ID: %s", id_)
+            # Get the timestamp from the ID
+            timestamp = int(timestamp_id.split("-")[0])
+            # Avoid fetching the same ID repeatedly
+            if id_ in fetched_ids:
+                continue
```
Comment (frankbaele) [inline app/_import.py:93]: add a debug log that the fetch is skipped

File: app/_import.py [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+        # We update the min_id to the last ID of the batch
+        min_id = batch[-1][0]
+        for timestamp_id, item in batch:
+            id_ = item[id_field_name]
+            logger.debug("Fetched ID: %s", id_)
+            # Get the timestamp from the ID
+            timestamp = int(timestamp_id.split("-")[0])
+            document = document_fetcher.fetch_document(stream_name, item)
+            if document is None:
+                continue
```
Comment (frankbaele) [inline app/_import.py:140]: add debug log that document is skipped

**Review by frankbaele — COMMENTED** [review]
looks good to me, its a nice clean up. Added 2 comments, some extra debugging for skipped records.

**Comment by raphael0202** [comment]
I've added extra log messages, thank you for your review!

**Review by frankbaele — APPROVED** [review]

**Review by frankbaele — APPROVED** [review]

**Pull request merged**